### PR TITLE
REF: Create numba helper function for jitting + generating cache key

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] xref https://github.com/pandas-dev/pandas/pull/33388#discussion_r413967963
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -77,11 +77,8 @@
 from pandas.core.series import Series
 from pandas.core.util.numba_ import (
     NUMBA_FUNC_CACHE,
```
Comment (jreback) [inline pandas/core/groupby/generic.py:79]: can you remove the CACHE from here as well?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/groupby/generic.py (hunk 1 above)
Comment (mroeschke) [reply pandas/core/groupby/generic.py:79]: There are some separate code paths that store the function in the cache independently

Comment (jreback) [reply pandas/core/groupby/generic.py:79]: i c. ideally this is not visible at all in this code if at all possible (this is what I mean by using a Dispatcher, e.g. that you would be how you would interactive with the function & cache itself). but i guess that's for the future.

**Review by jreback — APPROVED** [review]

File: pandas/core/groupby/generic.py (hunk 1 above)
Comment (mroeschke) [reply pandas/core/groupby/generic.py:79]: Ah I see. Okay I can look into that in a future PR when playing around more with a Dispatcher

**Pull request merged**